### PR TITLE
Improve documentation and semantics around internal buffer handling

### LINKED ISSUE #733: wants_read() returns true on new ClientSession
In rustls/src/client/mod.rs line 445, the client session wants to read before the handshake has started. However, in the TLS protocol, there is never anything to read before the client sent the initial hello.

### PR DESCRIPTION
**Pull request by djc**
Fixes #733, #734.

### CODE REVIEW

File: rustls/src/conn.rs [hunk 1]
```diff
@@ -835,8 +837,16 @@ impl ConnectionCommon {
         rc
     }
 
-    pub fn has_readable_plaintext(&self) -> bool {
-        !self.received_plaintext.is_empty()
+    pub fn wants_read(&self) -> bool {
+        // We want to read more data all the time, except when we have unprocessed plaintext.
+        // This provides back-pressure to the TCP buffers. We also don't want to read more after
+        // the peer has sent use a close notification.
```
Comment (jsha) [inline rustls/src/conn.rs:843]:
```suggestion
        // the peer has sent us a close notification.
```

**Review by jsha — COMMENTED** [review]
One typo, otherwise looks good to me. Thanks!

File: rustls/src/conn.rs (hunk 1 above)
Comment (djc) [reply rustls/src/conn.rs]: Good catch, fixed!

**Comment by ctz** [comment]
👍🏽

**Pull request merged**